src/experimental/ProbabilisticGraphicalModels/bayesian_network.jl

```diff
@@ -10,7 +10,8 @@ struct BayesianNetwork{V,T,F}
     "mapping from variable names to ids"
     names_to_ids::Dict{V,T}
     "values of each variable in the network"
-    values::Dict{V,Any} # TODO: make it a NamedTuple for better performance in the future
+    evaluation_env::NamedTuple
+    loop_vars::NamedTuple
```

Comment:
`loop_vars` is different for each variable, so should be a dict.

For instance,
```julia
@bugs begin
    for i in 1:3
        x[i] ~ Normal()
    end
end
```
`x[1]` would have `loop_vars = (i=1,)`, `x[2]` would have `loop_vars = (i=2,)`, etc.